### DISCUSSION TITLE
Manually add-certificate works but how can I check that the auto renew is working?

### DISCUSSION THREAD

**Discussion by luu-at-pharm3r** [shibayan/appservice-acmebot#291] [Q&A] [answered]
Hi There,
Thanks for your function app, it's saved us having to do all of this manually and paying the big corps.
Manually using the function app and /add-certificate works to setup the initial certs but now after 2+ months, how can I check that the auto renew is working as it's inside 30 day renewal period and my certificates are not auto renewing? What logs can I view or tests can I do to verify the issue?
Thanks,
Luu

**Comment by shibayan** [comment]
The automatic update process currently runs every Monday at midnight.

You can check the execution logs by selecting RenewCertificates from the Application Insights or Function list. Setting up a webhook is useful for notifying Slack and Teams of the success or failure of an update.

**Comment by luu-at-pharm3r** [comment, marked as answer]
Thanks for pointing to the error logs, I can see some in application insights. I have a 2 errors each function call Order, IssueCertificate, RenewCertificates_SubOrchestrator endpoints. The error for order is not very detailed can you give some insight to what to fix... "Message | Exception while executing function: Order NewOrder request did not specify any identifiers" where do I setup the identifiers and what are all the letsencrpt variables needed? Is the manual renewal (initial setup) different than the auto renewal configs?
Thanks,
Luu

**Reply by shibayan** [reply]
It could be a bug, so I've updated Acmebot now, please try again after restarting Azure Function. I think some kind of warning log will be output.

**Reply by luu-at-pharm3r** [reply]
Got it, made the update. How can I run the weekend (Monday midnight) auto renew manually to see the logs?
Thanks,
Luu

**Reply by shibayan** [reply]
You can run it manually by opening `RenewCertificates_Timer` from Azure Functions. Make sure that the name of the function to be executed is correct.

![image](https://user-images.githubusercontent.com/1356444/109542790-e19c3e00-7b08-11eb-9714-423ad8e16ad2.png)

It is easier to open `RenewCertificates_SubOrchestrator` and check the execution log from Monitor. It should be even easier to see if you open the Orchestrator tab.

![image](https://user-images.githubusercontent.com/1356444/109543061-393aa980-7b09-11eb-8859-2001cc4340bb.png)

**Reply by luu-at-pharm3r** [reply]
Hi there, I ran it manually and looks like it was generating fine based on logs. But I think what happens is that it generates a new private key certificate (I have a few now from testing) but my app service is binded to the soon-to-be expired certificate in the TLS/SSL Settings. I have to manually change the binding in order to get the newly generated certificate. This seems to happen only for wildcard certificates (I have an app at subdomain.mydomain.com and a wildcard SSL certificate applied for *.mydomain.com). Attached image for reference... any advice to get it to switch out automatically instead of having to switch the bindings manually? 
![image](https://user-images.githubusercontent.com/67646015/109769196-41115100-7bd0-11eb-9574-0c87a8b78a31.png)

Thanks for your help.

**Reply by shibayan** [reply]
Thanks for the feedback. There seems to be a bug where the bindings are not updated for wildcards. This will be fixed as soon as possible.

**Reply by shibayan** [reply]
Did you manually bind a wildcard certificate to a non-wildcard hostname? Or are you using a wildcard certificate that is shared with another App Service?

**Reply by luu-at-pharm3r** [reply]
The hostname is api.mydomain.com and I binded the wildcard certificate *.mydomain.com to it yes. This wildcard certificate is used with other app services as well... so www. and staging. subdomains also use *.mydomain.com wildcard certificate. Thanks for covering different scenarios.

**Reply by shibayan** [reply]
Such a scenario of manually setting up certificate binding is not supported. This means that App Service Acmebot cannot share certificates with multiple App Services.

In such a scenario, you need to use the Key Vault version.
https://github.com/shibayan/keyvault-acmebot

**Reply by luu-at-pharm3r** [reply]
OK thanks... I'll look at implementing the keyvault one. Thanks for your support.
